packages/password-field/src/vaadin-password-field.js

```diff
@@ -0,0 +1,195 @@
+    return '22.0.0-alpha1';
+  }
+
+  static get template() {
+    if (!memoizedTemplate) {
+      // Clone the superclass template
+      memoizedTemplate = super.template.cloneNode(true);
+
+      // Retrieve this element's dom-module template
+      const revealButton = ownTemplate.content.querySelector('[part="reveal-button"]');
+
+      // Append reveal-button and styles to the text-field template
+      const inputField = memoizedTemplate.content.querySelector('[part="input-field"]');
+      inputField.appendChild(revealButton);
+    }
+
+    return memoizedTemplate;
+  }
+
+  static get properties() {
+    return {
+      /**
+       * Set to true to hide the eye icon which toggles the password visibility.
+       * @attr {boolean} reveal-button-hidden
+       */
+      revealButtonHidden: {
+        type: Boolean,
+        observer: '_revealButtonHiddenChanged',
+        value: false
+      },
+
+      /**
+       * True if the password is visible ([type=text]).
+       * @attr {boolean} password-visible
+       */
+      passwordVisible: {
+        type: Boolean,
+        value: false,
+        reflectToAttribute: true,
+        observer: '_passwordVisibleChanged',
+        readOnly: true
+      }
+    };
+  }
+
+  get slots() {
+    return {
+      ...super.slots,
+      reveal: () => {
+        const btn = document.createElement('button');
+        btn.setAttribute('type', 'button');
+        btn.setAttribute('tabindex', '0');
+        return btn;
+      }
+    };
+  }
+
+  /** @protected */
+  get _revealNode() {
+    return this._getDirectSlotChild('reveal');
+  }
+
+  constructor() {
+    super();
+    this._setType('password');
+    this.__boundRevealButtonClick = this._onRevealButtonClick.bind(this);
+    this.__boundRevealButtonTouchend = this._onRevealButtonTouchend.bind(this);
+  }
+
+  /** @protected */
+  ready() {
+    super.ready();
+
+    this._revealPart = this.shadowRoot.querySelector('[part="reveal-button"]');
+  }
+
+  /** @protected */
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (this._revealNode) {
+      this._revealNode.setAttribute('aria-label', 'Show password');
+      this._revealNode.addEventListener('click', this.__boundRevealButtonClick);
+      this._revealNode.addEventListener('touchend', this.__boundRevealButtonTouchend);
+    }
+
+    if (this._inputNode) {
+      this._inputNode.autocapitalize = 'off';
+    }
+
+    this._toggleRevealHidden(this.revealButtonHidden);
+    this._updateToggleState(false);
+  }
+
+  /** @protected */
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    if (this._revealNode) {
+      this._revealNode.removeEventListener('click', this.__boundRevealButtonClick);
+      this._revealNode.removeEventListener('touchend', this.__boundRevealButtonTouchend);
+    }
+  }
+
+  /**
+   * @param {Event} e
+   * @return {boolean}
+   * @protected
+   */
+  _shouldRemoveFocus(e) {
+    return e.relatedTarget !== this._revealNode;
```

Comment:
The hide password on focus out feature is not working as it did in the old version.

### Issue: Password hides when focusing input

**Steps:**
1. enter password
2. click reveal button
3. click on input with password

**Old behaviour:**
- password stays visible
- after step 2. the input either still has focus, or receives focus

**New behaviour:**
- password is hidden again
- after step 2. the input is not focused

Only happens in desktop mode /w click events. In mobile simulator it works like the old behavior, except that the input is not focused.

Tested with: Chrome, FF

### Issue: Focus out on reveal button inconsistent on mobile

**Steps:**
1. enter password
2. click reveal button
3. click outside of password field
4. click reveal button
5. click outside of password field

**Old behaviour:**
- both after step 3. and 5. the password is hidden

**New behaviour:**
- after step 3. the password is hidden
- after step 5. the password is still visible

Only happens in mobile simulator /w touch events.

Tested with: Chrome, FF

Reply:
Another small inconsistency related to the focus ring:
- tabbing from input to reveal button keeps focus ring around input
- back-tabbing to the reveal button does not add a focus ring around the input

Reply:
> Issue: Password hides when focusing input

Fixed.

> back-tabbing to the reveal button does not add a focus ring around the input

Fixed.

> after step 2. the input is not focused

This is expected behavior, see https://github.com/vaadin/web-components/pull/2275#issuecomment-895038139

Regarding the issue on mobile, I will continue investigating and will let you know when I push a fix for it.

Reply:
> after step 5. the password is still visible

Updated to restore `focus()` call on the input on `touchend` event, so now this is fixed.